src/storage/src/hummock/compactor/mod.rs

```diff
@@ -294,6 +261,10 @@ impl Compactor {
                 context.sstable_store.delete_cache(table.id);
             }
         }
+        context
```

Comment:
without `scopeguard::guard` , shall we need to guarantee `remove_watermark_sst_id`  is called on error handling ？